lua/nvim-treesitter/ts_utils.lua

```diff
@@ -259,14 +259,14 @@ function M.update_selection(buf, node, selection_mode)
   selection_mode = selection_mode or "charwise"
   local start_row, start_col, end_row, end_col = M.get_vim_range({ M.get_node_range(node) }, buf)
 
-  vim.fn.setpos(".", { buf, start_row, start_col, 0 })
-
   -- Start visual selection in appropriate mode
   local v_table = { charwise = "v", linewise = "V", blockwise = "<C-v>" }
   ---- Call to `nvim_replace_termcodes()` is needed for sending appropriate
   ---- command to enter blockwise mode
   local mode_string = vim.api.nvim_replace_termcodes(v_table[selection_mode] or selection_mode, true, true, true)
   vim.cmd("normal! " .. mode_string)
+  vim.fn.setpos(".", { buf, start_row, start_col, 0 })
+  vim.cmd "normal! o"
```

Comment:
and it's possible to use `o` here and all other selection mode like `V` and `<C-v>` keep working? I don't know very much about this area of Vim and would trust you. Could we add a test? We would need tests anyway when upstreaming this function to Neovim

Maybe suggest your change also here: https://github.com/neovim/neovim/pull/18232/

Reply:
@theHamsta I added some small tests (charwise, linewise and blockwise). There are not many functional tests so wasn't fully sure about the desired format. Also, is there some description somewhere for how to run the tests locally? I tried just running `scripts/run_tests.sh` but I need to setup eg plenary locally etc. So I haven't actually executed the tests that I added yet, but hopefully they will pass.

Reply:
Woops, the new tests didn't actually run. I didn't put them in a file ending with `_spec`. Moved them to `ts_utils_spec.lua` now instead.